ui/components/core/src/components/Histogram.svelte

```diff
@@ -15,105 +15,81 @@
-        cell: {
-          stroke: "transparent",
-        },
-      },
-    },
-    params: [
-      {
-        name: "highlight",
-        select: {
-          type: "point",
-          on: "mouseover",
-        },
+  // Separate splits
+  let splits: string[] = [...new Set(hist.histogram.map((item) => String(item.split)))];
+  let datasets: ChartDataset[] = [];
+  for (let i = 0; i < splits.length; ++i) {
+    datasets.push({
+      label: splits[i],
+      backgroundColor: colors[i] + "40",
+      borderColor: colors[i],
+      borderWidth: 2,
+      data: hist.histogram.filter((item) => item.split === splits[i]).map((item) => item.counts),
+    });
+  }
+
+  // Define label names
+  let labels;
+  if (hist.type == "numerical")
+    labels = [...new Set(hist.histogram.map((item) => item.bin_start + "-" + item.bin_end))];
+  else if (hist.type == "categorical")
+    labels = [...new Set(hist.histogram.map((item) => item[hist.name]))];
+
+  // Prepare chart data and options
+  let data = {
+    labels: labels,
+    datasets: datasets,
+  };
+  let options = {
+    maintainAspectRatio: false,
+    plugins: {
+      title: {
+        display: true,
+        text: hist.name,
       },
-    ],
-    data: {
-      values: hist.histogram,
-    },
-    height: h,
-    width: 150,
-    background: "transparent",
-    mark: { type: "bar", tooltip: true },
-    encoding: {
-      x: {
-        field: "counts",
-        type: "quantitative",
-        axis: {
-          title: null,
-          grid: false,
-          domain: false,
-          ticks: false,
-          labels: false,
+      zoom: {
+        zoom: {
+          wheel: {
+            enabled: true,
+            speed: 0.05,
+          },
+          mode: "x",
         },
-      },
-      y: {
-        field: hist.type == "categorical" ? hist.name : "bin_range",
-        type: "nominal",
-        sort: "x",
-        axis: {
-          title: null,
-          domain: false,
-          grid: false,
-          ticks: false,
-          labelLimit: 50,
-          labelPadding: 10,
+        pan: {
+          enabled: true,
+          mode: "x",
         },
       },
-      color: {
-        condition: [
-          {
-            param: "highlight",
-            empty: false,
-            value: "grey",
-          },
-        ],
-        field: "split",
-      },
+    },
+    responsive: true,
+    scales: {
+      x: { stacked: true },
+      y: { stacked: true },
     },
   };
 
-  if (hist.type == "numerical") {
-    // Set bin ranges
-    for (let i = 0; i < hist.histogram.length; ++i) {
-      const value = hist.histogram[i];
-      value.bin_range = value.bin_start + "-" + value.bin_end;
-    }
-  }
-
-  // Vega-lite chart options
-  const options = {
-    actions: false,
-  };
+  Chart.register(Title, Tooltip, Legend, BarElement, CategoryScale, LinearScale, zoomPlugin);
 </script>
 
 <!-- Histogram -->
-<div class="max-h-{maxHeight} w-72 mx-auto flex flex-col justify-center items-center">
```

Comment:
`max-h-{maxHeight}` won't work here because of how tailwind parses values https://tailwindcss.com/docs/content-configuration#class-detection-in-depth

if you really need dynamic values, you can use css variables instead:
```
  <div
    class="..."
    style="max-height:{maxHeight}"
  >
```

Reply:
This line was deleted and dynamic values are no longer needed.

Reply:
yep i read too quickly!